Source/WebCore/workers/WorkerRunLoop.cpp

```diff
@@ -133,8 +133,10 @@ class RunLoopSetup {
         : m_runLoop(runLoop)
         , m_isForDebugging(isForDebugging)
     {
-        if (!m_runLoop.m_nestedCount)
+        if (!m_runLoop.m_nestedCount) {
+            m_runLoop.m_sharedTimer = makeUnique<WorkerSharedTimer>();
```

Comment:
Why did you move the construction here instead of the constructor?

Reply:
Oh because the constructor runs in the main thread so thread identity check will fail otherwise.